__tests__/allTests.test.js

```diff
@@ -0,0 +1,233 @@
+const app = require('../app')
+const request = require('supertest')
+const db = require('../db/connection')
+const seed = require('../db/seeds/seed')
+const testData = require('../db/data/test-data')
+const endpoints = require('../endpoints.json')
+
+beforeEach(() => seed(testData))
+
+afterAll(() =>db.end())
+
+
+
+describe('/api/topics', () => {
+    test('GET:200 sends an object of topics to the client', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) =>{
+            //console.log(body)
+            expect(body.topics).toHaveLength(3)
+            body.topics.forEach((topic) => {
+               // console.log(topic)
+
+                expect(topic).toMatchObject({
+                    slug: expect.any(String),
+                    description: expect.any(String)
+                })
+            })
+
+        });
+    })
+})
+
+
+// I think we should have had api call all other
+describe('/api/', () => {
+    test('GET:200 sends an object with API endpoints to the client', async () => {
+      const response = await request(app).get('/api/').expect(200);
+      const body = response.body;
+  
+     // console.log(body);
+
+      expect(body).toBeInstanceOf(Object);
+  
+      expect(body).toHaveProperty('GET /api');
+      expect(body).toHaveProperty('GET /api/articles');
+      expect(body).toHaveProperty('GET /api/articles/:article_id');
+      expect(body).toHaveProperty('GET /api/topics');
+  
+      expect(body['GET /api']).toMatchObject({
+        description: expect.any(String),
+      });
+  
+      expect(body['GET /api/articles']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+
+      //console.log(body['GET /api/articles/:article_id'])
+  
+      expect(body['GET /api/articles/:article_id']).toMatchObject({
+        article_id: expect.any(String),
+        article_img_url: expect.any(String),
+        author: expect.any(String),
+        created_at: expect.any(String),
+        title: expect.any(String),
+        topic: expect.any(String),
+        votes: expect.any(Number),
+      });
+  
+        //console.log(body['GET /api/topics'])
+
+      expect(body['GET /api/topics']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+  
+      expect(body['GET /api/topics'].exampleResponse.topics).toBeInstanceOf(Array);
+      body['GET /api/topics'].exampleResponse.topics.forEach((topic) => {
+        expect(topic).toMatchObject({
+          slug: expect.any(String),
+          description: expect.any(String),
+        });
+      });
```

Comment:
This works, but every time you add a new endpoint, you'll have to add to it again! You can help make this test more dynamic by requiring in the endpoints.json file directly, then expecting the response from your server to be equal to it. This will keep the test dynamic!